src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -683,6 +685,23 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
+        internal async Task FlushAsync() // no cancellation token: we don't want individual operation tokens applying to the whole connection
+        {
+            if (NetEventSource.IsEnabled) Trace("");
+            await AcquireWriteLockAsync(default).ConfigureAwait(false);
+            try
+            {
```

Comment:
Related to my comment below: I think we need to check for _inProgressWrite here and await it. Otherwise we may do two writes.